src/core/syscalls/os_syscall_handler.rs

```diff
@@ -1,2 +1,944 @@
+pub struct TransactionExecutionInfo;
+
+impl OsSingleStarknetStorage {
+    // Writes the given value in the given key in ongoing_storage_changes and returns the
+    // previous value. This value is needed to create the DictAccess while executing the
+    // corresponding storage_write system call.
+    fn write(&self, _key: u32, _value: u32) -> u32 {
+        // TO BE IMPLEMENTED
+        todo!()
+    }
+}
+
+#[derive(Debug)]
+struct OsSyscallHandler {
+    tx_execution_info_iterator: VecDeque<TransactionExecutionInfo>,
+    call_iterator: VecDeque<CallInfo>,
+
+    //  A stack that keeps track of the state of the calls being executed now.
+    // The last item is the state of the current call; the one before it, is the
+    // state of the caller (the call the called the current call); and so on.
+    call_stack: VecDeque<CallInfo>,
+    // An iterator over contract addresses that were deployed during that call.
+    deployed_contracts_iterator: VecDeque<u32>, // U32
+    // An iterator to the retdata of its internal calls.
+    retdata_iterator: VecDeque<VecDeque<u32>>, //VEC<U32>
+    // An iterator to the read_values array which is consumed when the transaction
+    // code is executed.
+    execute_code_read_iterator: VecDeque<u32>, //u32
+    // StarkNet storage members.
+    starknet_storage_by_address: HashMap<u32, OsSingleStarknetStorage>,
+    // A pointer to the Cairo TxInfo struct.
+    // This pointer needs to match the TxInfo pointer that is going to be used during the system
+    // call validation by the StarkNet OS.
+    // Set during enter_tx.
+    tx_info_ptr: Option<Relocatable>,
+    // The TransactionExecutionInfo for the transaction currently being executed.
+    tx_execution_info: Option<TransactionExecutionInfo>,
+}
+
+impl OsSyscallHandler {
+    #[allow(clippy::too_many_arguments)]
+    fn new(
+        tx_execution_info_iterator: VecDeque<TransactionExecutionInfo>,
+        call_iterator: VecDeque<CallInfo>,
+        call_stack: VecDeque<CallInfo>,
+        deployed_contracts_iterator: VecDeque<u32>,
+        retdata_iterator: VecDeque<VecDeque<u32>>,
+        execute_code_read_iterator: VecDeque<u32>,
+        starknet_storage_by_address: HashMap<u32, OsSingleStarknetStorage>,
+        tx_info_ptr: Option<Relocatable>,
+        tx_execution_info: Option<TransactionExecutionInfo>,
+    ) -> Self {
+        Self {
+            tx_execution_info_iterator,
+            call_iterator,
+            call_stack,
+            deployed_contracts_iterator,
+            retdata_iterator,
+            execute_code_read_iterator,
+            starknet_storage_by_address,
+            tx_info_ptr,
+            tx_execution_info,
+        }
+    }
+
+    // Called when starting the execution of a transaction.
+    // 'tx_info_ptr' is a pointer to the TxInfo struct corresponding to said transaction.
+    fn start_tx(&mut self, tx_info_ptr: Relocatable) -> Result<(), SyscallHandlerError> {
+        match self.tx_info_ptr {
```

Comment:
I think this could look cleaner using the `if let` syntax:

```rust
if let Some(_) = self.tx_info_ptr {
 return Err(SyscallHandlerError::ShouldBeNone(String::from("tx_info_ptr")));
}
```